src/lib/common/globals.h

```diff
@@ -174,12 +174,10 @@ typedef enum ApiVersion
 
 
 /* ****************************************************************************
-* Future date to represent permanent subscriptions.
-* High enough to make the subscription "permanent" but leaving room for
-* some (sloppy) increments, without causing overflow and accidental subscription
-* inactivation.
+*
+* PERMANENT_EXPIRES_DATETIME - date for permanent subscriptions/registrations
```

Comment:
For some, 0x7FFF could be criptic. A comment could help, eg.:

> We use as permanent the large possible value for 64 bits integer, which is millions of year in the future

However, this raise a concern... currently expiration is defined as `long long`. As far as I remember, basic numeric types size is implementation specific in C.

Should be better define expires as int_64t type? Could cause problems (i.e. how the field is stored or processed in MongoDB)? Style says something about this? (I have a vague memory about it...)

Should be

Reply:
No problem in using `int64_t` instead of `long long`.  It is the same, of course.
And yes, I think the style guide says to use the stdint types (int64_t, et al).
However, this sometimes gives problems with overloaded functions in the mongo client library.

If 0x7FFF... is cryptic, I am sure there is a MAXLONGLONG defined somewhere, and if not, we can always define it ourselves and do this:

```
#define PERMANENT_EXPIRES_DATETIME   MAXLONGLONG
```
Would this remove your convern?

Reply:
Fixrd in 88d49e6

Reply:
Fixed in 88d49e6

Reply:
By the moment is ok... but if we change in some moment `expires` to `int64_t` this should be adjusted (as LLONG_MAX could not be the same that "max int 64bits" depending on how long is implemented for particular C implementations).

Reply:
Not sure what you are worried about ...
long long and int64_t is the exact same thing. I strongly doubt you will find any machine not fulfilling this. Definitely not a linux distro

Reply:
NTC